### PR TITLE
fixed product reviews by product code route by api convention

### PR DESCRIPTION
**Pull request by antonioperic**

### CODE REVIEW

**Comment by mamazu** [comment]
Thank you for your contribution and welcome to the community. Your pull request looks great.
Could you please add an entry to the `UPGRADE.md` file for the changes?  This would provide a good way to notice the route changes when upgrading.

File: src/Resources/config/routing/productByCode.yml [hunk 1]
```diff
@@ -21,3 +9,15 @@ sylius_shop_api_add_product_review_by_code:
     methods: [POST]
     defaults:
         _controller: sylius.shop_api_plugin.controller.product.add_review_by_code_action
+
```
Comment (mamazu) [inline src/Resources/config/routing/productByCode.yml]: Why was that moved?

Comment (antonioperic) [reply src/Resources/config/routing/productByCode.yml]: /reviews is more specific route so it is moved on top, cause in some situaion routing can consider /reviews as part of slug/code
in this situation that will never happen
and also it is better/faster to have more specific routes on top

Comment (mamazu) [reply src/Resources/config/routing/productByCode.yml]: Yeah, that makes sense.

**Comment by mamazu** [comment]
Thank you, Antonio! :tada:
